parlai/tasks/image_chat/agents.py

```diff
@@ -240,14 +186,111 @@ def next_example(self):
         else:
             return ready
 
-    def share(self):
+    def share(self) -> TShared:
         shared = super().share()
         shared['data'] = self.data
         shared['image_loader'] = self.image_loader
         shared['personalities'] = self.personalities
         return shared
 
 
+class GenerationTeacher(ImageChatTeacher):
+    """
+    GenerationTeacher - dialogues are split into two episodes, one from
+    each individual person's point of view.
+
+    Used in the #dodecaDialogue task. (see https://parl.ai/projects/dodecadialogue/)
+    """
+
+    def __init__(self, opt: Opt, shared: TShared = None):
+        if opt.get('datasplit', '100k') != '100k':
+            print('Warning: teacher only has 100k datasplit')
```

Comment:
`warn_once` ?

Reply:
actually this is an artifact from internal, removing